### PR TITLE
S3415: Add FN/FP repro for #6630

### PR DESCRIPTION
**Pull request by cristian-ambrosini-sonarsource**
Test for #6630.

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/AssertionArgsShouldBePassedInCorrectOrder.Xunit.cs [hunk 1]
```diff
@@ -27,3 +27,20 @@ public void Foo()
...
+    class Program
+    {
+        [Fact]
+        public void Foo()
+        {
+            var str = "";
+            Assert.Equal(actual: "", expected: str); // Compliant FN
+            Assert.Equal(expected: "", actual: str); // Compliant
+            Assert.Equal(actual: str, expected: ""); // Noncompliant FP
+            Assert.Equal(expected: str, actual: ""); // Noncompliant
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/AssertionArgsShouldBePassedInCorrectOrder.Xunit.cs:43]: Do we support all the "Equals" variants described here?
https://xunit.net/docs/comparisons
* Equal
* NotEqual
* NotSame
* Same
for xunit and the corresponding methods for the other frameworks. If so, we should add more test cases (the parameter names might differ) and if not we need another issue.

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/AssertionArgsShouldBePassedInCorrectOrder.Xunit.cs (hunk 1 above)
Comment (cristian-ambrosini-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/AssertionArgsShouldBePassedInCorrectOrder.Xunit.cs:43]: For both **MsTest** and **NUnit** we support:
- `AreEqual`
- `AreNotEqual`
- `AreSame`
- `AreNotSame`
For **XUnit**:
- `Equal`
- `Same`
- `NotSame`

The rule is checking whether MemberAccessExpressionSyntax expression name is one of the above so in order to keep the test cases as minimal as possible I think we can avoid adding those as well, WDYT?

Comment (cristian-ambrosini-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/AssertionArgsShouldBePassedInCorrectOrder.Xunit.cs:43]: You are right indeed. I'll add those test cases as well.

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**